engine/storage/src/main/java/org/apache/cloudstack/storage/allocator/AbstractStoragePoolAllocator.java

```diff
@@ -185,6 +185,11 @@ public boolean configure(String name, Map<String, Object> params) throws Configu
             pools = reorderPoolsByCapacity(plan, pools);
         }
 
+        if (vmProfile.getVirtualMachine() == null) {
+            s_logger.trace("The VM is null, skipping pools reordering by disk provisioning type.");
+            return pools;
+        }
+
         if (vmProfile.getHypervisorType() == HypervisorType.VMware &&
```

Comment:
I think what @DaanHoogland meant was extending this to: `if (vmProfile.getVirtualMachine() != null && ... ` and we won't need the lines above